### DISCUSSION TITLE
is it / will it be possible to read/write data externally?

### DISCUSSION THREAD

**Discussion by NafeeJ** [ppy/osu#25276] [Q&A] [answered]
no idea how mongodb realms works but was just wondering if its possible or there are plans to make it possible to read/write things like scores, replays, beatmaps, or anything externally. at the very least there are programs like osu trainer and mapping tools which rely on read/write access.

obviously it wont be as direct but maybe there is/would be some nuget package, API, or whatever to make that possible?

**Comment by peppy** [comment, marked as answer]
External apps touching osu! files have never been condoned, and have resulted in issues for us over the years (people report breakage to us as if it's our fault).

They will not be condoned with lazer.
